### ISSUE TITLE
Dragon cannot easily operate the details element

### ISSUE THREAD

**Issue by selfthinker** [alphagov/govuk-design-system#4029]
This issue is from the accessibility audit of the Design System website as well as the accessibility audit of GOV.UK Frontend components by DAC in July 2024.

## DAC's description

* Page: [Type scale](https://design-system.service.gov.uk/styles/type-scale/) and [Details](https://design-system.service.gov.uk/components/details/)
* WCAG reference: none (usabililty)
* Issue ID: DAC_Details_Component_Usability_01 (website) and DAC_Usability_Detail_Component_01 (Frontend)

The ‘View the old type scale’ details component could not be accessed with Dragon using the expected voice commands. This put voice activation users at a disadvantage of as they cannot access the component using the expected voice commands. The use of the `<details>` and `<summary>` components contain some browser and assistive technology issues which means that it is not well supported. Voice activation users were still able to access the component using advanced voice commands, but this is not expected behaviour.

<img width="745" alt="Screenshot of the details component in action" src="https://github.com/user-attachments/assets/c072ef2f-769b-47c4-9610-d5c0390705b6">

### Voice Activation user comments

“I am unable to interact with ‘View the old type scale’ element using standard Dragon commands. I tried the direct Dragon Command using its visual title ‘click view the old type scale’ but this did not work. I then tried using the ‘click link’, ‘click button’, and ‘click list’ commands which gives Dragon tags to all possible links, buttons, and lists Dragon can identify on-screen respectively.
Unfortunately, no Dragon tags appeared on the element. This showed that Dragon could not identify this element as something that could be interacted with. The only way I was able to open the element was to use Dragon to navigate via keyboard control commands: ‘press Tab’ and the ‘press Enter’, which replicates the effects of pressing the Tab and the Enter keyboard buttons respectively.
This required lots more commands to open the element, making the process much longer and far more tedious. I had expected the element to be interactable via Dragon command based off the visual title ‘click view the old type scale’. This would allow me to open the element and view its contents in a consistent, quick, and easy manner.”

“The expandable component does not respond to Dragon commands. I am unable to open the ‘GOV.uk one login’ with the command ‘click button’ or by label by name. This means I will have to use keyboard commands through dragon to be able to expand the component to read the further information available. To do this I will use the command ‘press Tab’ to move my focus onto the component, then the command press enter to activate it. This is extremely frustrating and time consuming to do as there is many components to tab through to active it.
I would expect the expandable component to activate when I use the command ‘click about Gov.uk one login’ this make it more accessible and faster to access.”

## DAC's proposed solution

We suggest following the current implementation of the [GOV.UK Design System accordion component](https://design-system.service.gov.uk/components/accordion/) to replicate the behaviour of the details component.
The style of the component can remain the same, however by following the accordion component it will be implemented as a standard HTML button which can be accessed as expected by voice activation users using the expected voice commands.
Another suggestion would be to have the information readily available on the page, instead of it being contained within an expandable section.

## Other potential solutions

This is a known issue with Dragon. This happens with all native HTML `<details>` elements, irrespective of implementation.
 
We don't agree that changing the implementation to imitate the accordion component is the best solution.
We need to discuss how to fix this in a better way.
We might need to change the guidance to refer to the known issues.
Another option is to lobby for making Nuance fix the bug in Dragon.

## Related issues

* https://github.com/alphagov/govuk-frontend/issues/3693
* https://github.com/alphagov/govuk-design-system-backlog/issues/44
* https://github.com/alphagov/govuk-frontend/issues/3464

## Additional instances

Although the details component is used on multiple pages throughout the website, it is always the same component, so will only need to be fixed once.

## Needed roles

Frontend developer, content designer

**Comment by matteason** [comment]
I happened to test this out today while investigating some other Dragon issues and to my surprise it worked:

https://github.com/user-attachments/assets/4d123643-8ac6-489a-a4ec-2c4210409de0

I'm not quite sure how or why it's working. I've been digging into the [Dragon Web Extension](https://chromewebstore.google.com/detail/dragon-web-extension/ddaloccgjfibfpkalenodgehlhkgoahe?hl=en) and that doesn't have `<summary>` in its hardcoded list of clickable elements. But it seems to even work without the extension enabled, which is very unusual (most elements don't work properly or are very flaky)

Dragon Home 15.30.200.017, Chrome 130.0.6723.92, Windows 11 Home 23H2

**Comment by selfthinker** [comment]
We have decided not to fix this.

We thoroughly [investigated this](https://github.com/alphagov/govuk-frontend/issues/5308).
We found only one way to make it work for Dragon and that is to add the `role="button"` to the `<summary>` element. But doing that created further problems for some screen readers. Those could all potentially be fixed but to the detriment of needing to polyfill the whole component, which means additional JavaScript for everyone. And it would also mean the way screen readers handle it would be inconsistent.

We [asked DAC for their opinion](https://github.com/alphagov/govuk-design-system/issues/4194) on this, and they agreed that it's not worth doing due to not just current but also future problems with adding ARIA attributes to the component. They also said Dragon users are "not currently adversely impacted" because most Dragon users are used to the Details components not working properly and would therefore automatically use the known workarounds.

Another potential solution is to build our own component. But we don't think that would be a good idea either.
And another idea was to [create a browser extension to fix this](https://github.com/alphagov/govuk-frontend/issues/5448). We might pursue this in the future.

**Comment by selfthinker** [comment]
I have summarised all the [technical details around the details element in this JSbin](https://jsbin.com/nubebij), including testable code.
